src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,626 @@
+// mam4xx: Copyright (c) 2022,
+// Battelle Memorial Institute and
+// National Technology & Engineering Solutions of Sandia, LLC (NTESS)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef MAM4XX_AERO_MODEL_HPP
+#define MAM4XX_AERO_MODEL_HPP
+
+#include <ekat/util/ekat_math_utils.hpp>
+
+#include <haero/atmosphere.hpp>
+#include <haero/math.hpp>
+
+#include <mam4xx/aero_config.hpp>
+#include <mam4xx/conversions.hpp>
+#include <mam4xx/mam4_types.hpp>
+#include <mam4xx/utils.hpp>
+
+namespace mam4 {
+
+namespace aero_model {
+
+// BAD CONSTANT
+const Real dlndg_nimptblgrow = haero::log(1.25);
+constexpr int nimptblgrow_mind = -7, nimptblgrow_maxd = 12;
+constexpr int nimptblgrow_total = -nimptblgrow_mind + nimptblgrow_maxd + 1;
+const int nrainsvmax = 50; // maximum bin number for rain
+const int naerosvmax = 51; //  maximum bin number for aerosol
+const int maxd_aspectype = 14;
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_bcscavcoef_get(
+    const int imode, const bool isprx_kk, const Real dgn_awet_imode_kk, //& ! in
+    const Real dgnum_amode_imode,
+    const Real scavimptblvol[nimptblgrow_total][AeroConfig::num_modes()],
+    const Real scavimptblnum[nimptblgrow_total][AeroConfig::num_modes()],
+    Real &scavcoefnum_kk, Real &scavcoefvol_kk) {
+
+  // !-----------------------------------------------------------------------
+  // ! compute impaction scavenging removal amount for aerosol volume and number
+  // !-----------------------------------------------------------------------
+
+  // @param [in]  imode mode index
+  // @param [in]  isprx_kk if there is precip
+  // @param [in]  dgn_awet_imode_kk ! wet aerosol diameter of mode imode at
+  // elevation kk [m]
+  // @param [out] scavcoefnum scavenging removal for aerosol number [1/h]
+  // @param [out] scavcoefvol scavenging removal for aerosol volume [1/h]
+
+  // NOTE: original FORTRAN function has two internal loops: kk and icol.
+  // We removed these loops. Hence, the inputs/outputs of
+  // modal_aero_bcscavcoef_get are reals at kk , icol locations
+
+  const Real zero = 0;
+  const Real one = 1;
+  if (isprx_kk) {
+    // ! interpolate table values using log of
+    // (actual-wet-size)/(base-dry-size) ratio of wet and dry aerosol diameter
+    // [fraction]
+    const Real wetdiaratio = dgn_awet_imode_kk / dgnum_amode_imode;
+    // BAD CONSTANT
+    Real scavimpvol, scavimpnum = zero;
+    if (wetdiaratio >= 0.99 && wetdiaratio <= 1.01) {
+      scavimpvol = scavimptblvol[-nimptblgrow_mind][imode];
```

Comment:
Indexing for `scavimptblnum` in C++ $\in$ [0,19]. Indexing for scavimptblnum in Fortran $\in$ [-7,12]. Hence, for this array, index 0 in Fortran will be 7 (`-nimptblgrow_mind`) in C+++. In both cases, (0 or 7) will be the 8th element in scavimptblnum.
Here I am assuming that `nimptblgrow_mind` is negative.

Reply:
I didn't know you could use simple Latex math expressions in GitHub!

@odiazib , it may be worth adding a comment describing the indexing scheme as you have above, just so people aren't confused later on.